new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Remove magic and method_missing from Chef::Resource lookup
+
+This proposal:
+
+- Switches to explicit methods for Chef recipe DSL instead of method_missing
+- Removes all special treatment of classes in the Chef::Resource and Chef::Provider namespaces
+- Automatically adds resource DSL for all named descendants of Chef::Resource, no matter what their namespace
```

Comment:
Also because I have to say it, "This won't be thread safe" but I don't think anyone cares about multi-threading the initial cookbook loading process yet, so as long as we all remember it should be fine :-)